glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt

```diff
@@ -112,14 +121,20 @@ class PingUploadWorker(context: Context, params: WorkerParameters) : Worker(cont
                         action.request.body,
                         action.request.headers,
                         Glean.configuration
-                    )
+                    ).toFfi()
+
+                    if (result == Constants.UPLOAD_RESULT_RECOVERABLE ||
+                        result == Constants.UPLOAD_RESULT_UNRECOVERABLE) {
```

Comment:
A thing I only notice now:

* We know that `unrecoverable` failures will lead to the file being deleted, we would only need to count the recoverable errors here.
  * HTTP status 400-499 are also unrecoverable, at least internally to the Rust part. Again, we don't need to care, the ping is deleted and not retried.
* android-components never uses `UPLOAD_RESULT_UNRECOVERABLE` anyway (there we catch all I/O errors as `recoverable`

If we do fail on I/O in Fenix we correctly catch that here with the `recoverable` part.